cypress/integration/CTS/SearchResultsPageTests/index.js

```diff
@@ -0,0 +1,189 @@
+});
+
+And('{string} pager is displayed', (pagerPosition) => {
+    cy.get(`div.results-page__control.--${pagerPosition} .results-page__pager`).should('be.visible');
+});
+
+And('result list is displayed', () => {
+    cy.get('div.results-list').should('be.visible');
+});
+
+And('each result item is a link', () => {
+    cy.document().then((doc) => {
+        const titleLocator = doc.querySelectorAll('.results-list-item__title');
+        for (let i = 0; i < titleLocator.length - 1; i++) {
+            cy.get(titleLocator[i]).should('be.visible');
+            cy.get(titleLocator[i]).find('a').should('have.attr', 'href');
+        }
+    });
+});
+
+And('there are {int} delighters present', (count) => {
+    cy.get('.cts-delighter-container div[class^="delighter"]:visible').should('have.length', count);
+});
+
+And('the following delighters are displayed', (dataTable) => {
+    for (const { delighter, href, title, text } of dataTable.hashes()) {
+        cy.get(`div[class="delighter ${delighter}"]:visible`).as('delighter');
+        cy.get('@delighter').find('a').first().should('have.attr', 'href', href);
+        cy.get('@delighter').find('h4').first().should('have.text', title);
+        cy.get('@delighter').find('p').first().should('have.text', text);
+    }
+});
+
+And('chat pop up is displayed', () => {
+    cy.wait(30000);
+    cy.get('#ProactiveLiveHelpForCTSPrompt').should('be.visible');
+});
+
+And('chat pop up title is {string}', (chatTitle) => {
+    cy.get('#ProactiveLiveHelpForCTSPrompt').find('h2').should('contain', chatTitle);
+});
+
+And('{string} button has a link {string}', (chatNowButton, link) => {
+    cy.get('#proactive-chat-form').should('have.attr', 'action', link);
+    cy.get('#chat-button').should('have.text', chatNowButton);
+});
+
+
+When('user clicks on Chat Now button', () => {
+    cy.get('form').then((form$) => {
+        form$.on('submit', (e) => {
+            e.preventDefault();
+            cy.get('button#chat-button').click();
+        });
+    });
+});
+
+Then('the chat is opened in a new tab', () => {
+    cy.get('#proactive-chat-form').should('have.attr', "target", "_blank");
+});
+
+And(`all result item's checkboxes are not checked`, () => {
+    cy.get('.results-list .cts-checkbox .cts-checkbox__label').each(($el) => {
+        const win = $el[0].ownerDocument.defaultView;
+        // use getComputedStyle to read the pseudo selector
+        const before = win.getComputedStyle($el[0], ':before');
+        // read the value of the `content` CSS property
+        const contentValue = before['background-image'];
+        // the returned value will have double quotes around it, but this is correct
+        expect(contentValue).to.eq('none');
+    });
+});
+
+And('{string} checkbox is displayed', (selectAllCheckbox) => {
+    cy.get(`div.cts-checkbox.check-all label[for='select-all-cbx-top']`).should('be.visible').and('have.text', selectAllCheckbox);
```

Comment:
this locator seems very long - i think you can get away with using either div or label

Reply:
done